File: remove_bg.js
const Jimp = require('jimp');

async function removeWhiteBg(inputPath, outputPath) {
  try {
    const image = await Jimp.read(inputPath);
    
    // Iterate over every pixel
    image.scan(0, 0, image.bitmap.width, image.bitmap.height, function(x, y, idx) {
      const red = this.bitmap.data[idx + 0];
      const green = this.bitmap.data[idx + 1];
      const blue = this.bitmap.data[idx + 2];
      
      // If the pixel is close to white, make it transparent
      if (red > 230 && green > 230 && blue > 230) {
        this.bitmap.data[idx + 3] = 0; // alpha to 0
      }
    });

    await image.writeAsync(outputPath);
    console.log('Processed:', outputPath);
  } catch (err) {
    console.error('Error processing', inputPath, err);
  }
}

async function main() {
  await removeWhiteBg('./public/images/floral_top.png', './public/images/floral_top.png');
  await removeWhiteBg('./public/images/floral_bottom.png', './public/images/floral_bottom.png');
  console.log('Done');
}

main();

File: remove_bg.py
from PIL import Image
import sys

def remove_white(img_path):
    img = Image.open(img_path).convert("RGBA")
    datas = img.getdata()

    newData = []
    for item in datas:
        # If pixel is close to white, make it transparent
        if item[0] > 200 and item[1] > 200 and item[2] > 200:
            newData.append((255, 255, 255, 0))
        else:
            newData.append(item)

    img.putdata(newData)
    img.save(img_path, "PNG")

remove_white(sys.argv[1])
